src/test/java/dk/cngroup/lentils/repository/UserRepositoryTest.java

```diff
@@ -0,0 +1,72 @@
+package dk.cngroup.lentils.repository;
+
+import dk.cngroup.lentils.LentilsApplication;
+import dk.cngroup.lentils.entity.Team;
+import dk.cngroup.lentils.entity.User;
+import dk.cngroup.lentils.service.ObjectGenerator;
+import dk.cngroup.lentils.service.TeamService;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
+
+@Transactional
+@RunWith(SpringRunner.class)
+@SpringBootTest(classes = {LentilsApplication.class, ObjectGenerator.class})
+public class UserRepositoryTest {
+    @Autowired
+    TeamService teamService;
+
+    @Autowired
+    UserRepository userRepository;
+
+    @Test
+    public void saveTeamCreateNewUser() {
+        long count = userRepository.count();
+        Team team = new Team("teamA", 5, "1234");
+        User user = new User();
+        addValidUserToTeam(team, user);
+        teamService.save(team);
+        Assert.assertEquals(count + 1, userRepository.count());
+    }
+
+    @Test
+    public void adminUserIsImportedOnStartup() {
+        Assert.assertEquals(1, userRepository.count());
+        List<User> users = userRepository.findAll();
+        Assert.assertEquals("admin",users.get(0).getUsername());
+    }
+
+    @Test
+    public void updateTeamDoNotCreateNewUser() {
+        Team team = new Team("teamA", 5, "1234");
+        User user = new User();
+        addValidUserToTeam(team, user);
+        teamService.save(team);
+        long count = userRepository.count();
+        teamService.update(team);
+        Assert.assertEquals(count, userRepository.count());
+    }
+
+    @Test
+    public void deleteTeamRemoveUser() {
+        Team team = new Team("teamA", 5, "1234");
+        User user = new User();
+        addValidUserToTeam(team, user);
+        teamService.save(team);
+        long count = userRepository.count();
+        teamService.delete(team.getTeamId());
+        Assert.assertEquals(count - 1, userRepository.count());
+    }
+
+    private void addValidUserToTeam(Team team, User user) {
+        user.setUsername("testUsername");
```

Comment:
`user.setUsername("testUsername");` a `user.setPassword("pwd");` by tady nemusely byt. Toto muze byt prave v tom ObjectGenerator kldine. Muzes tam mit metodu `createUser`, `createTeam`, `createTeamWithUser`, ... ckoli potrebujes. Ale pokud uz tam toho bude vic, tak by bylo fojn mit tady ten object generator pro ruzne entitni typy oddeleny. Proc se mi to tady nelibi je to, ze se metoda jmenuje `addValidUserToTeam` ale z tooto nazvu nepoznas, ze se tam jeste do usera setuje username a password a az pak se prida do teamu. Idealni by bylo: 1. vytvorit usera (s username, password) 2. vytvorit team, 3. priradit uzivatele do tymu ;)

Reply:
vytvoreny zvlast metody pro jednotlive akce - createUser, createTeam, addUserToTeam